src/nvim/option.c

```diff
@@ -3354,10 +3354,6 @@ did_set_string_option(
     if (!parse_winhl_opt(curwin)) {
       errmsg = e_invarg;
     }
-  } else if (varp == &p_rtp) {  // 'runtimepath'
-    if (!nlua_update_package_path()) {
-      errmsg = (char_u *)N_("E970: Failed to initialize lua interpreter");
-    }
```

Comment:
You're still going to make sure that when we update rtp, the Lua paths get updated immediately? (and in the same Lua evaluation?)

Reply:
they are updated immediately per definition as the separate representation has been erased.